Common/Thread/Promise.h

```diff
@@ -45,6 +45,7 @@ class PromiseTask : public Task {
 template<class T>
 class Promise {
 public:
+	// Never fails.
```

Comment:
(as long as you assume OOMs don't exist or exist as a separate catastrophe than failure.)

-[Unknown]

Reply:
True, but we're kinda screwed in several other ways simultaneously then. That said, "Never fails" is always a bit of an overconfident comment to put - but the meaning here is that there's no need to check for null return.